server/ai_process.go

```diff
@@ -1048,13 +1048,14 @@ func submitLiveVideoToVideo(ctx context.Context, params aiRequestParams, sess *A
 		if err != nil {
 			return nil, fmt.Errorf("invalid control URL: %w", err)
 		}
-		// TODO any errors from these funcs should we kill the input stream?
-
 		events, err := common.AppendHostname(*resp.JSON200.EventsUrl, host)
 		if err != nil {
 			return nil, fmt.Errorf("invalid events URL: %w", err)
 		}
 		clog.V(common.VERBOSE).Infof(ctx, "pub %s sub %s control %s events %s", pub, sub, control, events)
+
+		params.node.LivePipelines[params.liveParams.stream] = &core.LivePipeline{}
+
 		startTricklePublish(ctx, pub, params, sess)
 		startTrickleSubscribe(ctx, sub, params)
 		startControlPublish(control, params)
```

Comment:
Nice catch! I wonder if we can achieve the same effect by calling `startControlPublish` before `startTricklePublish` without having to change anything else?

Reply:
Yeah you're right that will work too, done 👍